py4DSTEM/process/diskdetection/diskdetection_cuda.py

```diff
@@ -671,39 +644,38 @@ def dftUpsample(imageCorr, upsampleFactor, xyShift):
             Correlation image between two images in Fourier space.
         upsampleFactor (int):
             Scalar integer of how much to upsample.
-        xyShift (list of 2 floats):
+        xyShift (N_points,2) cp.ndarray, locations to upsample around:
             Coordinates in the UPSAMPLED GRID around which to upsample.
             These must be single-pixel IN THE UPSAMPLED GRID
 
     Returns:
         (ndarray):
-            Upsampled image from region around correlation peak.
+            Stack of upsampled images from region around correlation peak.
     """
+    N_pts = xyShift.shape[0]
     imageSize = imageCorr.shape
     pixelRadius = 1.5
-    numRow = np.ceil(pixelRadius * upsampleFactor)
-    numCol = numRow
-
-    colKern = cp.exp(
-        (-1j * 2 * cp.pi / (imageSize[1] * upsampleFactor))
-        * cp.outer(
-            (cp.fft.ifftshift((cp.arange(imageSize[1]))) - cp.floor(imageSize[1] / 2)),
-            (cp.arange(numCol) - xyShift[1]),
-        )
-    )
+    kernel_size = int(np.ceil(pixelRadius * upsampleFactor))
 
-    rowKern = cp.exp(
-        (-1j * 2 * cp.pi / (imageSize[0] * upsampleFactor))
-        * cp.outer(
-            (cp.arange(numRow) - xyShift[0]),
-            (cp.fft.ifftshift(cp.arange(imageSize[0])) - cp.floor(imageSize[0] / 2)),
-        )
-    )
+    colKern = cp.zeros((N_pts, imageSize[1], kernel_size),dtype=cp.complex64) # N_pts * image_size[1] * kernel_size
+    rowKern = cp.zeros((N_pts, kernel_size, imageSize[0]),dtype=cp.complex64) # N_pts * kernel_size * image_size[0]
+
+    # Fill in the DFT arrays using the CUDA kernels
+    multicorr_col_kernel = kernels["multicorr_col_kernel"]
+    blocks = ((np.prod(colKern.shape) // multicorr_col_kernel.max_threads_per_block + 1),)
+    threads = (multicorr_col_kernel.max_threads_per_block,)
+    multicorr_col_kernel(blocks,threads,(colKern, xyShift, N_pts, *imageSize, upsampleFactor))
 
+    multicorr_row_kernel = kernels["multicorr_row_kernel"]
+    blocks = ((np.prod(rowKern.shape) // multicorr_row_kernel.max_threads_per_block + 1),)
+    threads = (multicorr_row_kernel.max_threads_per_block,)
+    multicorr_row_kernel(blocks,threads,(rowKern, xyShift, N_pts, *imageSize, upsampleFactor))
+
+    # Apply the DFT arrays to the correlation image
     imageUpsample = cp.real(rowKern @ imageCorr @ colKern)
     return imageUpsample
 
-
+@numba.jit(nopython=True)
```

Comment:
I think this can be written as a `numba.stencil` function, which I believe is even faster. https://numba.pydata.org/numba-doc/latest/user/stencil.html. 

Could also consider setting `target='gpu'` ?

Reply:
I was not aware of stencil, that may come in handy in the future. However this function isn't called on the entire array so it doesn't make sense here. 

I'm not sure if it would be faster to come up with a CUDA solution for this particular interpolation operation or if launching a kernel for such a small unit of work would end up being slower than this rather quick CPU function, so I'm going to leave that for the future.